src/iocore/cache/StripeSM.cc

```diff
@@ -109,13 +109,9 @@ struct StripeInitInfo {
   }
 };
 
-StripeSM::StripeSM(CacheDisk *disk, off_t blocks, off_t dir_skip) : Continuation(new_ProxyMutex()), Stripe{disk, blocks, dir_skip}
+StripeSM::StripeSM(CacheDisk *disk, off_t blocks, off_t dir_skip)
+  : Continuation(new_ProxyMutex()), Stripe{disk, blocks, dir_skip}, _preserved_dirs{static_cast<int>(len)}
```

Comment:
What's the right way to do this in C++? This makes me nervous because it depends on initialization order.

Reply:
I've added a comment explaining the weirdness.